posthog/apps.py

```diff
@@ -42,6 +47,17 @@ def ready(self):
             if settings.SERVER_GATEWAY_INTERFACE == "WSGI":
                 async_to_sync(initialize_self_capture_api_token)()
 
+            # This runs for ALL Django startups in DEBUG mode
+            try:
+                logger.info("Waiting for plugin-server to be ready...")
+                if wait_for_plugin_server():
```

Comment:
Let's move this to a celery task as then we don't block the setup if you arent running the plugin server locally.